Topics/Development_Process/Technical_Documents/Intro_to_rfcs.md

```diff
@@ -0,0 +1,96 @@
+# **Introduction to RFCs**
+
+The following is an introductory overview of RFCs based on the following 2 articles by Juan Pablo Buriticá:
+
+1. ["A thorough team guide to RFCs"](https://leaddev.com/technical-decision-making/thorough-team-guide-rfcs)
+2. ["6 Lessons I learned while implementing technical RFCs as a decision making tool"](https://buriti.ca/6-lessons-i-learned-while-implementing-technical-rfcs-as-a-management-tool-34687dbf46cb)
+
+## **What are RFCs?**
+
+- RFC stands for Request for Comment
+- It is a document that acts as a technical proposal for a design, approach or change
+- It can outline things like architecture changes, dependencies, assumptions, alternative considerations, etc.
+- RFCs are meant to facilitate technical decision-making and increase visibility and collaboration within an organization
+- RFCs are usually authored by a few people and then circulated around to receive feedback (hence the name “request for comments”)
+- An RFC can be understood as “Here is the direction we want to go, are we missing anything?”
+- There are 4 general phases in an RFC life cycle
+  - Proposal stage: the RFC is drafted
+  - Commenting stage: feedback and discussion ensue
+  - Decision stage: ideally, the reviewers and the author align on the path forward
+  - Archive stage: the RFC is archived for future reference
+
+## **Why are RFCs useful?**
+
+- Mitigates the risk of the decision made
+  - RFCs act as a kind of code review but at the planning stage
+  - Problems or considerations can be surfaced before resources have been spent
+- Facilitates better technical decisions and collaboration
+  - Building software is a social exercise
+  - RFCs facilitates communication and decision-making between groups
+- Allows for individual contribution and responsibility on all levels
+  - When done right, individuals of all levels in the organization are empowered and encouraged to make impactful decisions while seeking input from others
+- Creates documentation for future context
+  - An archive of past RFCs act as a record of previous discussions and decisions
+- Is an asynchronous process
+  - Multiple RFCs can run in parallel
+  - Increases the flexibility of collaborative discussions
+
+## **When to RFC?**
+
+If it crosses your mind that an RFC might be required to make a change or implement a design, it probably is a good idea to make one.
+
+Additionally, an RFC is generally a good idea when:
+
+- You are building something from scratch (new endpoint, component, system, library, application, etc.).
```

Comment:
I think this can be another good reason why RFC can be helpful to other 301 students especially to teams who are building a new project from scratch!